libraries/ms-drivers/src/adt7476a_fan_controller.c

```diff
@@ -15,14 +15,22 @@
 // accepts number between 0-100, converts into into range of 0x00 - 0xFF
 StatusCode adt7476a_set_speed(I2CPort port, uint8_t speed_percent, AdtFanGroup fan_group,
                               uint8_t adt7476a_i2c_address) {
+  // check for out of range conditions.
+  if (speed_percent > 100) {
+    return STATUS_CODE_OUT_OF_RANGE;
+  }
+
   // determine which PWM output to change
-  uint8_t real_speed = floor(speed_percent / 0.39);
+  uint8_t real_speed = (speed_percent / 0.39);
   if (fan_group == ADT_FAN_GROUP_1) {
-    status_ok_or_return(i2c_write_reg(port, adt7476a_i2c_address, ADT7476A_PWM_1, &speed_percent,
+    status_ok_or_return(i2c_write_reg(port, adt7476a_i2c_address, ADT7476A_PWM_1, &real_speed,
                                       SET_SPEED_NUM_BYTES));
-  } else {
-    status_ok_or_return(i2c_write_reg(port, adt7476a_i2c_address, ADT7476A_PWM_3, &speed_percent,
+
+  } else if (fan_group == ADT_FAN_GROUP_2) {
+    status_ok_or_return(i2c_write_reg(port, adt7476a_i2c_address, ADT7476A_PWM_3, &real_speed,
                                       SET_SPEED_NUM_BYTES));
+  } else {
+    return STATUS_CODE_INVALID_ARGS;
   }
```

Comment:
so it turns out we're going to be using this fan controller in other boards as well and will need to control up to 4 fans with it. Can we generalize this to all three PWM outputs?